include/solvers/source_terms.h

```diff
@@ -50,6 +50,7 @@ namespace SourceTerms
       : navier_stokes_source(dim + 1)
       , heat_transfer_source(1)
       , tracer_source(1)
+      , cahn_hilliard_source(1)
```

Comment:
It should be 2 if my understanding is right
```suggestion
      , cahn_hilliard_source(2)
```